sql/src/main/java/io/crate/operation/collect/sources/NodeStatsCollectSource.java

```diff
@@ -115,4 +104,29 @@ public boolean apply(DiscoveryNode input) {
             )
         );
     }
+
+    @Nullable
+    private List<String> nodeIds(WhereClause whereClause) {
+        if (whereClause.hasQuery()) {
+            final List<List<Symbol>> exactMatches = this.eqExtractor.extractParentMatches(
+                ImmutableList.<ColumnIdent>builder()
```

Comment:
I think you could use `ImmutableList.of(ID)` here to avoid the builder allocation